File: src/main/java/fr/antod3v/plugins/pterodactyl/extension/Pterodactyl.java
package fr.antod3v.plugins.pterodactyl.extension;

import lombok.Getter;
import lombok.Setter;
import org.gradle.api.Action;
import org.gradle.api.Project;

@Getter
public class Pterodactyl {

    private final Project project;

    private final Credential credential;

    private final Deploy deploy;

    public Pterodactyl(Project project) {
        this.project = project;
        this.credential = new Credential();
        this.deploy = new Deploy(project);
    }

    public void credential(Action<? super Credential> action) {
        action.execute(credential);
    }

    public void deploy(Action<? super Deploy> action) {
        action.execute(deploy);
    }

}

File: src/main/kotlin/fr/antod3v/plugins/pterodactyl/Pterodactyl.kt
package fr.antod3v.plugins.pterodactyl

import fr.antod3v.plugins.pterodactyl.extension.Credential
import fr.antod3v.plugins.pterodactyl.task.DeployTask
import fr.antod3v.plugins.pterodactyl.task.ExecTask
import org.gradle.api.Plugin
import org.gradle.api.Project

class Pterodactyl : Plugin<Project> {

    override fun apply(project: Project) {

        val credential = project.extensions.create("pterodactyl", Credential::class.java, project)

        project.tasks.create("deploy", DeployTask::class.java) {

            it.apiUrl.set(credential.apiUrl)
            it.apiKey.set(credential.apiKey)
            it.serverId.set(credential.apiKey)

        }

        project.tasks.create("exec", ExecTask::class.java) {

            it.apiUrl.set(credential.apiUrl)
            it.apiKey.set(credential.apiKey)
            it.serverId.set(credential.apiKey)

        }

    }

}